src/adcm_client/objects.py

```diff
@@ -917,7 +917,10 @@ def provider(self) -> "Provider":
 
     def cluster(self) -> "Cluster":
         """Return 'Cluster' object"""
-        return self._parent_obj(Cluster)
+        if self.cluster_id:
```

Comment:
I think it's better to explicitly use the check `is None`. You can also get rid of `else`.

Reply:
Done in next commit